diesel/src/mysql/connection/raw.rs

```diff
@@ -142,6 +138,35 @@ impl RawConnection {
             ))
         }
     }
+
+    fn flush_pending_results(&self) -> QueryResult<()> {
+        self.consume_current_result()?;
+        while self.next_result()? {
+            self.consume_current_result()?;
+        }
+        self.consume_current_result()?;
```

Comment:
We just talked about this, so I know that for some reason this needs to be used in this way, but otherwise I'd have guessed that at least one of those calls to `self.consume_current_result` was a copypasta error… Can you add a short comment citing some mysql doc/source and say that this is on purpose?